### DISCUSSION TITLE
MIT License Compatible Screenpack?

### DISCUSSION THREAD

**Discussion by EVILEDACCOUNT** [ikemen-engine/Ikemen-GO#838] [Feature Request]
Hello. Work on this engine has been going well and it looks amazing. I play it a lot and really do enjoy it. I check almost daily for new builds and keep playing them to see new features and test things that I see and report back on feature ideas.

A little background. I love Free Software and have donated to the Free Software Foundation in the past. I believe in their mission and ethical philosophy. I have long, long, long wanted a MUGEN replacement engine that was licensed as Free Software as MUGEN itself is propitiatory. I finally had this dream fulfilled with this project. I contacted Suehiro along with ShinLucho to try to get a clear license assigned to the project. And, I was so happy when Suehiro and this project derived from his decided to use the MIT Expat License.

That all said, this project still uses a modified Screenpack from MUGEN that is fundamentally not license compatible. Which is why, I assume, it's properly stored in its own separate repository.

My vision is for a completely pure IKEMEN GO that has no dependencies that are license incompatible.

So, I am curious on what would be the best way to commission the creation of an entirely Free Software MIT Expat compatible screenpack that can be packaged with IKEMEN GO to replace the MUGEN derived screenpack.

Here is what I am thinking.

1: The fonts themselves should not be a problem to get around. All we should have to do is find good looking fonts under a compatible license that we can use in place of the MUGEN fonts. Maybe I've naive. But, there are fonts out there under these licenses and this shouldn't be a real problem.

2: We need to coordinate a clear theme for the, well, theme. I'm open to suggestions; but I think the blue and white color that the current IKEMEN GO icon uses would be a good idea. It's a striking theme and would look great.

As a side note; I'm currently thinking of making an IKEMEN GO arcade cabinet and am thinking of making the side art and marque use this theme of blue and white. Before I do that, and before we go further with the screenpack, is this a color theme you all have decided to stay with? Personally, I really like it and hope so. But, it would be good to know before proceeding. Also, what font did you use for the "I" for in the IKEMEN GO icon? Or, was this "I" custom drawn?

3: We need to ditch the MUGEN name entirely with this screenpack. This is IKEMEN GO, not MUGEN. It is a MUGEN replacement. But, I think everyone knows this already. It's kind of sad, as a pride thing, to be running IKEMEN GO; and every time I look at the main menu I see MUGEN in big letters and IKEMEN GO is in small letters off to the bottom left. This project is at this point, or will be soon, where it should really stand on its own name and not be tied to MUGEN in name anymore. Both for pride and for legal reasons as well.

This project is well known on the MUGEN'S Fighter's guild in the appropriately named 'beyond MUGEN' section. People know what this engine is for. It doesn't need the MUGEN name anymore in its screenpack.

4: We need to decide if we are going to call this project I.K.E.M.E.N. or I.K.E.M.E.N. GO. Both are fine. But, this should be decided before the creation of the screenpack. Suehiro himself seems to have ditched IKEMEN in favor of IKEMEN GO. The "GO" clearly distinguishes the two projects. Yet, the old project is basically no long being developed. So, should we keep the "GO" or just assume everyone knows it's "GO" and drop the "GO?" We should have clear "branding" with our name.

5: Once we decide if it should be called I.K.E.M.E.N. or I.K.E.M.E.N. GO; we should place this decided on name front and center in big letters on the main menu where "M.U.G.E.N." currently is sitting.

So, here is what I'm thinking all rolled together.

A Free Software MIT Expat compatible license screenpack with license compatible fonts that is built from the ground up with either I.K.E.M.E.N. or I.K.E.M.E.N. GO as it's clear name in big letters that uses the blue and white color theme throughout the main menu and all other menus that we can then package with this project.

Then, we can ditch the MUGEN derived theme and its repository forever and the project can all be under one license and functional without any license incompatible dependencies.

Sounds great to me. But, how do we achieve this?

Well, I don't know how to make a screenpack. But, as luck has it, many talented people do.

So, here is what I am thinking. We should hire someone to make this. We should go onto the Mugen Fighter's Guild and source talent from our community. We should then pay someone something fair to both sides to create this screenpack for us that will look decent and fulfill the license requirements.

I am not rich. But, I do have a little bit on money. The screen pack doesn't need to be jaw dropping. It just needs to be license compatible, look decent, and be functional.

I am open to discussing this idea further and maybe funding the creation of this screenpack myself if we can negotiate a decent price with an artist.

I love this engine. I love Free Software. I want this project to be able to be released as one entity and stored and developed together without the need for a non-free repo and non-free dependencies.

Please get back to me with what you think of this idea and maybe we can get someone to make this happen at a fair price. Thank you for your time.

**Comment by EVILEDACCOUNT** [comment]
First test image mock up.
![TEST0](https://user-images.githubusercontent.com/51429077/80923179-76ab9a80-8d71-11ea-875e-12bda305c37b.JPG)
![TEST1](https://user-images.githubusercontent.com/51429077/80923358-c2ab0f00-8d72-11ea-9df5-e18794e1ce11.JPG)

Second mock up; white stripe added and logo moved to fit.

![TEST2](https://user-images.githubusercontent.com/51429077/80927215-be3f2000-8d8b-11ea-90da-2f4960a7f1ee.JPG)

**Comment by K4thos** [comment]
I think it's a good idea to replace copyrighted Elecbyte screenpack and lifebar with new one down the road. Your proposed design also looks interesting.

But before any of this happen first this [compatibility issue with 4:3 stages](https://mugenguild.com/forum/msg.2468820) needs to solved (it's the only reason why we're still using lowres winmugen screenpack by default instead of mugen 1.0 720p one). I wasn't able to fix it myself and I don't think other devs are working on solving it currently, so don't expect transition to default 720p screenpack soon (and creating new screenpack for 4:3 aspect ratio in this day and age doesn't seem like a good idea).

Your effort with mockup and offer to commission work to artist is appreciated.

**Comment by LeoKids** [comment]
Keep in mind that all assets are licensed under CC-BY-NC,this means it can be used freely,as long as your using for non-commercial projects.

**Comment by EVILEDACCOUNT** [comment]
> as long as your using for non-commercial projects.

That actually highlights one of its key problems. Everything else about the project is Free Software and thus able to be used for commercial use, aside from this screenpack. The engine needs a screenpack to run at all (standard, not in itself a bad things.) I haven't tried for a very long time, but hopefully the engine is compatible with more screenpacks now than it used to be. As I did spends days trying different screenpacks and usually the fight bars would be sized wrong with almost every screenpack I tried. So, there was a long while (again haven't tried for some time so this might still be the case) where using the default screenpack was the easiest option to get the engine to run at all.

This is also not helped by the fact that even trying to find alternative screenpacks which are licensed under a Free Software License is very difficult. This is because for a very long time the MUGEN community unfortunately didn't really care about licensing. They just made stuff and put it out there without any license at all. From a strict legal perspective, this makes all of that content unusable in a commercial venture as everything put out without a license (completely irrelevant to the fact of whether or not source code is provided) is by default proprietary. I can't find the FSF page right now that says this, but it is true. It was a US Law passed that forced everything into being copyrighted to its author by default instead of going into the public domain even if no copyright was filed. A big part of this was because Dawn of the Dead went instantly into the public domain because they forgot to put a copyright on the movie. So, this law was passed to prevent any future movie (or for that matter software or anything else that could be copyrighted) from accidentally fallowing into the public domain ever again. This is why if you release a project, even with its source code, even if you intend it to be Free Software or Open Source; absolutely none of that matters unless you put a license to it. No license = propitiatory copyright to the owner no matter what because of this law (at least in the US.) So, yes, MUGEN has loads and loads of screenpacks, almost all of which are completely unusable in a commercial project because they have no license.

It just makes sense to me, then, that the default screenpack also complement the option for commercial use that the engine itself does. Otherwise it would force people who want to make a commercial project to have to make their own screenpack or find a appropriately licensed one and hope it is compatible. That seems, to me, to be both an antithetical and prohibiting roadblock to the project.

Being able to sell a Free Software project is actually intended. [https://www.gnu.org/philosophy/selling.en.html](url) Actually, it was MAME's old license that for a long time stopped it from being considered a real Free Software project because is specifically banned commercial usage. This was before MAME finally dropped that requirement and transitioned to the GNU GPL which allows for commercial usage.

Another point to mention, is back when I originally made this post the default screenpack even said MUGEN on it instead of IKEMEN. So, not only did I want to fix the Freedom issue surrounding commercial use; I also wanted to establish a separate identity for this project that was not MUGEN. This project is IKEMEN, not MUGEN. The name was eventually changed in the default screenpack which fixes this problem to some degree. But, the blue cloud aesthetic is still the same as old MUGEN.

At the end of the day, this is an engine project and not an art project. People are going to use whatever screenpack they want and that's a good thing. But, as for that default screenpack bundled with the engine, at some point, (I think ideally before a 1.0 release) this really should be fixed.
